src/RarFile.cpp

```diff
@@ -492,14 +456,14 @@ ssize_t Read(void* context, void* lpBuf, size_t uiBufSize)
         ctx->inbuffer > MAXWINMEMSIZE - (ctx->head - ctx->buffer))
     {
       // invalid data returned by UnrarXLib, prevent a crash
-      XBMC->Log(ADDON::LOG_ERROR, "CRarFile::Read - Data buffer in inconsistent state");
+      kodi::Log(ADDON_LOG_ERROR, "CRarFile::Read - Data buffer in inconsistent state");
       ctx->inbuffer = 0;
     }
 
     if (ctx->inbuffer == 0)
       break;
 
-    ssize_t copy = std::min(static_cast<ssize_t>(ctx->inbuffer), uicBufSize);
```

Comment:
The static_cast seems to be required, at least while cross compiling for ARM (rpi) with gcc 7.1.x
```
vfs.rar/src/RarFile.cpp: In member function ‘virtual ssize_t CRARFile::Read(void*, void*, size_t)’:
vfs.rar/src/RarFile.cpp:466:54: error: no matching function for call to ‘min(int64_t&, ssize_t&)’
     ssize_t copy = std::min(ctx->inbuffer, uicBufSize);
                                                      ^
```

https://stackoverflow.com/questions/14508022/error-no-matching-function-for-call-to-minlong-unsigned-int-unsigned-int